data_warehouse/services/aggregate_data_service.py

```diff
@@ -207,13 +224,63 @@ def get_records_by_time_department(request_user, department_id, time):
         return records
 
     @classmethod
-    def coverage_statistics(cls, context):
-        '''to get coverage statistics data'''
+    def table_export(cls, context):
+        '''处理表格导出相关的请求'''
+        handlers = {
+            cls.TABLE_NAME_TRANING_HOURS_SUMMARY: 'traning_hours_statistics',
+            cls.TABLE_NAME_COVERAGE_SUMMARY: 'coverage_statistics',
+            cls.TABLE_NAME_TRANING_SUMMARY: 'trainee_statistics',
+            cls.TABLE_NAME_TRANING_FEEDBACK: 'training_feedback'
+        }
+        request = context.get('request', None)
+        if request is None:
+            raise BadRequest('错误的参数。')
+        table_type = int(request.GET.get('table_type'))
+        handler = handlers.get(table_type, None)
+        if handler is None:
+            raise BadRequest('未定义的表类型。')
+        handler_method = getattr(cls, handler, None)
+        return handler_method(context)
 
     @classmethod
     def training_hours_statistics(cls, context):
         '''to get training hours statistics data'''
 
+    @classmethod
+    def trainee_statistics(cls, context):
+        '''培训学时与工作量'''
+
+    @classmethod
+    def coverage_statistics(cls, context):
+        '''专任教师培训覆盖率'''
+        request = context.get('request', None)
+        program_id = context.get('program_id', None)
+        start_time = context.get('start_time', None)
+        end_time = context.get('end_time', None)
+        department_id = context.get('department_id', None)
+
+        records = CoverageStatisticsService.get_traning_records(
+            request.user, program_id, department_id, start_time, end_time)
+        users_qs = User.objects.filter(id__in=records.values_list(
+            'user', flat=True).distinct())
+        group_users_by_ages = CoverageStatisticsService.groupby_ages(users_qs)
+        group_users_by_depts = CoverageStatisticsService.groupby_departments(
+            users_qs)
+        group_users_by_titles = CoverageStatisticsService.groupby_titles(
+            users_qs)
+        grouped_records = {
```

Comment:
As I said, your data might be used by @oosprey , split the process of retrieving data and the process of formatting data into two functions might be a better reusable solution. FYI:
https://github.com/DLUT-SIE/TMSFTT-BE/blob/b4ac87c910dc0212e03479bddf0c4dde2274cfbf/data_warehouse/services/aggregate_data_service.py#L73-L78